src/implementation/Client/GRPCClient/crypto.ts

```diff
@@ -0,0 +1,178 @@
+  }
+
+  decrypt(opts: DecryptRequest): Promise<Duplex>;
+  decrypt(inData: Buffer | ArrayBuffer | ArrayBufferView, opts: DecryptRequest): Promise<Buffer>;
+  async decrypt(
+    arg0: Buffer | ArrayBuffer | ArrayBufferView | DecryptRequest,
+    opts?: DecryptRequest,
+  ): Promise<Duplex | Buffer> {
+    // Handle overloading
+    // If we have a single argument, assume the user wants to use the Duplex stream-based approach
+    let inData: Buffer | undefined;
+    if (opts === undefined) {
+      opts = arg0 as EncryptRequest;
+    } else {
+      // Throws if arg0 is not a supported type
+      inData = this.toArrayBuffer(arg0);
+    }
+
+    // Ensure required options are present
+    if (!opts) {
+      throw new Error(`Parameter 'opts' must be defined`);
+    }
+
+    // Create the gRPC stream
+    const client = await this.client.getClient();
+    const grpcStream = client.decryptAlpha1();
+
+    // Create a duplex stream that will send data to the server and read from it
+    const duplexStream = new DaprChunkedStream(grpcStream, pbDecryptRequest, (req) => {
+      // Disable the @typescript-eslint/no-non-null-assertion linter in this block because opts here is guaranteed to be non-nil (see the check above), but TS doesn't seem to believe that
+      /* eslint-disable @typescript-eslint/no-non-null-assertion */
+      const reqOptions = new pbDecryptRequestOptions();
+      reqOptions.setComponentName(opts!.componentName);
+      if (opts!.keyName) {
+        reqOptions.setKeyName(opts!.keyName);
+      }
+      req.setOptions(reqOptions);
+      /* eslint-enable @typescript-eslint/no-non-null-assertion */
+    });
+
+    // Process the data
+    return this.processStream(duplexStream, inData);
+  }
+
+  private toArrayBuffer(inData: Buffer | ArrayBuffer | ArrayBufferView | string | any): Buffer {
+    if (typeof inData == "string") {
+      return Buffer.from(inData, "utf8");
+    } else if (typeof inData == "object" && Buffer.isBuffer(inData)) {
+      return inData;
+    } else if (typeof inData == "object" && ArrayBuffer.isView(inData)) {
+      return Buffer.from(inData.buffer, inData.byteOffset);
+    } else if (typeof inData == "object" && inData instanceof ArrayBuffer) {
+      return Buffer.from(inData);
+    } else {
+      throw new Error(
+        `Invalid value for the inData parameter: must be a Buffer, an ArrayBuffer, an ArrayBufferView, or a string`,
+      );
+    }
+  }
+
+  private processStream(duplexStream: DaprChunkedStream<any, any>, inData?: Buffer): Promise<Duplex | Buffer> {
+    // If the caller did not pass data (as a Buffer etc), return the duplex stream and stop here
+    if (!inData) {
+      return Promise.resolve(duplexStream);
+    }
+
+    // Send the data to the stream and wait for the response
+    return new Promise((resolve, reject) => {
+      let data = Buffer.alloc(0);
+
+      // Add event listeners
+      duplexStream.on("data", (chunk: Buffer) => {
+        if (chunk.length > 0) {
```

Comment:
I typically do: `chunk?.length` as it ensures chunk exists and it has a length greater than 0

Reply:
Done thanks